dispatcher/queue.go

```diff
@@ -1,54 +1,59 @@
 package dispatcher
 
 import (
-	"container/list"
+	"container/heap"
 	"sync"
 )
 
 // A PriorityQueue implements heap.Interface and holds Items.
 type PriorityQueue struct {
-	jobs *list.List
+	jobs []*Job
 	mu   sync.Mutex
 }
 
 // Len returns the length of the priority queue
 func (pq *PriorityQueue) Len() int {
-	return pq.jobs.Len()
+	return len(pq.jobs)
+}
+
+func (pq *PriorityQueue) Swap(i, j int) {
+	pq.jobs[i], pq.jobs[j] = pq.jobs[j], pq.jobs[i]
+}
+
+func (pq *PriorityQueue) Less(i, j int) bool {
+	// We want Pop to give us the highest, not lowest, priority so we use greater than here.
+	return pq.jobs[i].Priority > pq.jobs[j].Priority
+}
+
+func (pq *PriorityQueue) Push(x any) {
+	item := x.(*Job)
+	pq.jobs = append(pq.jobs, item)
+}
+
+func (pq *PriorityQueue) Pop() any {
+	old := pq.jobs
+	n := len(old)
+	item := old[n-1]
+	old[n-1] = nil // avoid memory leak
+	pq.jobs = old[0 : n-1]
+	return item
 }
 
 //Enqueue queues the item in its correct position
 func (pq *PriorityQueue) Enqueue(job *Job) {
```

Comment:
Can we add some stats on how much this improves over the existing implementation?

Reply:
Done in https://github.com/newscred/webhook-broker/pull/134/commits/2fd7b9b15e0b83ee8244e51a56d70d20a4050149